### PR TITLE
GDPR: Module/Feature Privacy Links

### PR DESCRIPTION
**Pull request by jaswrks**
Ref: p1HpG7-50l-p2

Calypso: This PR adds "Privacy Information" links into each Jetpack module/feature.

<p align="center"></p>

---

<p align="center"></p>

---

<p align="center"></p>

---

### How to Test this PR

- In Calypso Live, under Settings in the Calypso Dashboard (https://calypso.live/?branch=add/gdpr-feature-privacy-links) check all of the icons next to Jetpack features to ensure they are working properly; i.e., that the Privacy Information link is there and that it leads to the appropriate Jetpack.com privacy section.
- Help to make sure I didn't miss any Jetpack features.

### Overview of Changes

- Adds a new `SupportInfo` component. See [README](https://github.com/Automattic/wp-calypso/tree/a92cdd900ff34d7fa702b93c7193b952cee38e92/client/components/support-info) in the [`client/components/support-info`](https://github.com/Automattic/wp-calypso/tree/a92cdd900ff34d7fa702b93c7193b952cee38e92/client/components/support-info) directory. The new component is used to display a support info icon. When clicked, it displays a popover with a description, a learn more link, and a privacy info link.

  This new component is a way of making what was `.site-settings__info-link-container` into a reusable component, and therefore easier to adjust globally.

- Removes all occurrences of `.site-settings__info-link-container` in favor of the new `SupportInfo` component.

- Updates CSS adjustments across a few components to change references from `.site-settings__info-link-container` into `.support-info`.

- Adds support info icons to all `JetpackModuleToggle` instances, covering every Jetpack module/feature. **Help:** Please make sure I didn't miss any Jetpack features.

### Also In Jetpack (`/wp-admin`)

See: https://github.com/Automattic/jetpack/pull/9199

### CODE REVIEW

File: client/my-sites/site-settings/seo-settings/site-verification.jsx [hunk 1]
```diff
@@ -308,22 +308,13 @@ class SiteVerification extends Component {
...
-										target="_blank"
-									>
-										{ translate( 'Learn more' ) }
-									</ExternalLink>
-								</InfoPopover>
-							</div>
-
+							<SupportInfo
+								text={ translate(
+									'Provides hidden tags needed to verify your' +
```
Comment (a8ci18n) [inline client/my-sites/site-settings/seo-settings/site-verification.jsx]: Hi! I've found a possible matching string that has already been translated **1** time:
`translate( 'Provides the necessary hidden tags needed to verify your WordPress site with various services.' )` *ES Score: 15*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

Comment (jaswrks) [reply client/my-sites/site-settings/seo-settings/site-verification.jsx]: 👍

**Comment by rickybanister** [comment]
This looks great @jaswrks

File: client/components/support-info/index.jsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+ * Internal dependencies
+ */
+import InfoPopover from 'components/info-popover';
+import ExternalLink from 'components/external-link';
+
+class SupportInfo extends Component {
+	static propTypes = {
+		text: PropTypes.string,
+		link: PropTypes.string,
+		privacyLink: PropTypes.oneOfType( [ PropTypes.string, PropTypes.bool ] ),
```
Comment (kwight) [inline client/components/support-info/index.jsx]: Is it necessary to introduce the boolean usage here? Is it enough to just have an empty string by default, and check for empty below?

**Comment by kwight** [comment]
This a really big PR to review (+355,-401); could we split it up? The new component could get its own PR, and then more smaller, easier-to-test PRs for the changes to each module?

**Comment by jaswrks** [comment]
@kwight writes...

> This a really big PR to review (+355,-401); could we split it up? The new component could get its own PR, and then more smaller, easier-to-test PRs for the changes to each module?

Sure. Happy to break it apart. That'll be around 20 or so PRs. Do I have that right?

**Comment by beaulebens** [comment]
Just checking on the progress of this, given it doesn't seem to have been touched in a while, and we're looking at the deadline in a couple of days. @jaswrks is this still happening?

**Comment by jaswrks** [comment]
I'm resolving the latest merge conflicts now. I'll go ahead and break it apart into 1) new component, 2) component updates that make use of the new component. 3)

Do we want to break it apart further? A separate PR for every component change would be around 20+ PRs. I could use some guidance on how to make this easier to test without creating too much confusion either. Either way, I don't see any reason it can't be ready to merge later today.

**Comment by kwight** [comment]
Oh, I don't think 20+ PRs is necessary; how about one for adding the `support-info` component, then a few others to add the new component to existing components? It'll just make each PR easier to see what's going on and to know what needs testing.

**Comment by jaswrks** [comment]
All merge conflicts resolved and I re-tested all of the links and confirmed they all point to the proper location. I've also broken this PR apart into four separate PRs.

- [Add `SupportInfo` component for GDPR Privacy Information links](https://github.com/Automattic/wp-calypso/pull/24985)
  - [GDPR Privacy Information links (batch 1)](https://github.com/Automattic/wp-calypso/pull/24987)
    - [GDPR Privacy Information links (batch 2)](https://github.com/Automattic/wp-calypso/pull/24988)
    - [GDPR Privacy Information links (batch 3)](https://github.com/Automattic/wp-calypso/pull/24989)

Please let me know if there's anything more I can do to help merge this in. Thanks! :-)

**Comment by jaswrks** [comment]
Closing in favor of:

- [Add `SupportInfo` component for GDPR Privacy Information links](https://github.com/Automattic/wp-calypso/pull/24985)
  - [GDPR Privacy Information links (batch 1)](https://github.com/Automattic/wp-calypso/pull/24987)
  - [GDPR Privacy Information links (batch 2)](https://github.com/Automattic/wp-calypso/pull/24988)
  - [GDPR Privacy Information links (batch 3)](https://github.com/Automattic/wp-calypso/pull/24989)

**Pull request closed without merging**